metacat-main/src/main/java/com/netflix/metacat/main/services/impl/TableServiceImpl.java

```diff
@@ -267,10 +361,12 @@ public TableDto deleteAndReturn(final QualifiedName name, final boolean isMView)
             }
         }
 
-        // Try to delete the table even if get above fails
+        final Set<ChildInfo> childInfos = parentChildRelMetadataService.getChildren(name);
+        if (childInfos != null && !childInfos.isEmpty()) {
+            throw new RuntimeException("Fail to drop " + name + " because it still has child table");
```

Comment:
It might help to print at least one of the child tables in the exception message.

Since the child tables are not included in get table response, and most users will not know how to use the special get child endpoint. So if they try to drop a table and it fails then the exception message can at least tell them which of the child tables exist that is blocking the drop.